src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/Hashing/XxHash128.cs

```diff
@@ -0,0 +1,452 @@
+                    return HashLength0To16(sourcePtr, length, (ulong)seed);
+                }
+
+                if (length <= 128)
+                {
+                    return HashLength17To128(sourcePtr, length, (ulong)seed);
+                }
+
+                if (length <= MidSizeMaxBytes)
+                {
+                    return HashLength129To240(sourcePtr, length, (ulong)seed);
+                }
+
+                return HashLengthOver240(sourcePtr, length, (ulong)seed);
+            }
+        }
+
+        /// <summary>Resets the hash computation to the initial state.</summary>
+        public override void Reset()
+        {
+            XxHashShared.Reset(ref _state);
+        }
+
+        /// <summary>Appends the contents of <paramref name="source"/> to the data already processed for the current hash computation.</summary>
+        /// <param name="source">The data to process.</param>
+        public override void Append(ReadOnlySpan<byte> source)
+        {
+            XxHashShared.Append(ref _state, source);
+        }
+
+        /// <summary>Writes the computed 128-bit hash value to <paramref name="destination"/> without modifying accumulated state.</summary>
+        /// <param name="destination">The buffer that receives the computed hash value.</param>
+        protected override void GetCurrentHashCore(Span<byte> destination)
+        {
+            Hash128 current = GetCurrentHashAsHash128();
+            WriteBigEndian128(current, destination);
+        }
+
+        private Hash128 GetCurrentHashAsHash128()
+        {
+            Hash128 current;
+
+            if (_state.TotalLength > MidSizeMaxBytes)
+            {
+                // Digest on a local copy to ensure the accumulators remain unaltered.
+                ulong* accumulators = stackalloc ulong[AccumulatorCount];
+                CopyAccumulators(ref _state, accumulators);
+
+                fixed (byte* secret = _state.Secret)
+                {
+                    DigestLong(ref _state, accumulators, secret);
+                    current = new Hash128(
+                        low64: MergeAccumulators(accumulators, secret + SecretMergeAccsStartBytes, _state.TotalLength * Prime64_1),
+                        high64: MergeAccumulators(accumulators, secret + SecretLengthBytes - AccumulatorCount * sizeof(ulong) - SecretMergeAccsStartBytes, ~(_state.TotalLength * Prime64_2)));
+                }
+            }
+            else
+            {
+                fixed (byte* buffer = _state.Buffer)
+                {
+                    current = HashToHash128(new ReadOnlySpan<byte>(buffer, (int)_state.TotalLength), (long)_state.Seed);
+                }
+            }
+
+            return current;
+        }
+
+#if NET7_0_OR_GREATER
+        /// <summary>Gets the current computed hash value without modifying accumulated state.</summary>
+        /// <returns>The hash value for the data already provided.</returns>
+        // [CLSCompliant(false)]
```

Comment:
commented out CLSCompliant attributes.  they were causing warnings as we don't build with CLS arguments to compiler.